strawberryfields/tdm/tdmprogram.py

```diff
@@ -471,6 +471,8 @@ def compile(self, *, device=None, compiler=None):
             # Third check: the parameters of the gates are valid
             # We will loop over the different operations in the device specification
 
+            num_symbolic_param = 0  # counts the number of symbolic variables, which are labelled consecutively by the context method
```

Comment:
the issue was line 512 (`local_p_vals = self.tdm_params[num_symbolic_param]`): before this change, the index was effectively reset to zero for every gate, thus picking the wrong parameter.